### PR TITLE
Adding inference db

### PR DESCRIPTION
**Pull request by Hsankesara**
### Changes

1.  changed restapi routes & updated inference output
3. Added function to save inference results in postgres db
4. Added project_id in the rest app and lag in starttime in model invocation endpoint
6. Refactored the wine quality detection model
7. Updated mlflow invocation return output 
8. Added Alias in all the models
9. Added metric option in the best model invocation
10. Error handling in model invocation endpoint

### CODE REVIEW

File: model_class/lung_study.py [hunk 1]
```diff
@@ -305,9 +308,23 @@ def preprocess_data(self, raw_data):
...
+    def create_return_obj(self, indexes, model_name, model_version, alias, inference_result):
+        dateTimeObj = dt.now(tz=None)
+        return_obj = pd.DataFrame.from_dict(indexes, orient="index", columns=["uid", "date", "pid"])
+        return_obj["invocation_result"] = [{"anamoly_detected": result} for result in inference_result]
+        return_obj["model_name"] = model_name
+        return_obj["model_version"] = model_version
+        return_obj["alias"] = alias
+        return_obj["timestamp"] = dateTimeObj.timestamp()
+        return_obj["window_size"] = self.window_size
+        return_obj['invocation_result'] = return_obj.invocation_result.map(self.dict2json)
```
Comment (yatharthranjan) [inline model_class/lung_study.py:329]: this will be stored as postgres json type, right?

Comment (Hsankesara) [reply model_class/lung_study.py:329]: Yes, the return object will be stored as the json type.

File: model-invocation-endpoint/main.py [hunk 2]
```diff
@@ -26,26 +26,27 @@ def get_model_version_info(name: str, version: int):
 #def train_new_version():
```
[the remark this answers is not in the document]
Comment (Hsankesara) [reply model-invocation-endpoint/main.py:26]: Yes, will remove it.

File: model-invocation-endpoint/mlflow_interface.py [hunk 3]
```diff
@@ -11,6 +11,10 @@
 from dataloader.postgres_pandas_wrapper import PostgresPandasWrapper
 from model_class import ModelClass
 import importlib
+from sqlalchemy.exc import DataError, DBAPIError
+from requests.exceptions import ConnectionError
+from botocore.exceptions import EndpointConnectionError, ClientError
+
 class MlflowInterface():
```
[the remark this answers is not in the document]
Comment (Hsankesara) [reply model-invocation-endpoint/mlflow_interface.py:18]: Sure, I will create an issue for now and will refactor it later.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                with:
+                    images: ${{ env.DOCKERHUB_IMAGE }}
+                    tags: |
+                        type=semver,pattern={{version}}
+                        type=semver,pattern={{major}}.{{minor}}
+            -   name: Build docker
+                uses: docker/build-push-action@v2
+                with:
+                    # Allow running the image on the architectures supported by openjdk:11-jre-slim
+                    platforms: linux/amd64,linux/arm64
```
[the remark this answers is not in the document]
Comment (Hsankesara) [reply .github/workflows/release.yml:47]: Yes, the build supports arm64 as well.

**Review by yatharthranjan — APPROVED** [review]
LGTM. thanks

**Pull request merged**